src/middlewares/import-project.middleware.js

```diff
@@ -134,23 +134,27 @@ export default (store: any) => (next: any) => (action: any) => {
 };
 
 const inferProjectType = json => {
-  const dependencyNames = Object.keys(json.dependencies);
-
-  if (dependencyNames.includes('gatsby')) {
-    return 'gatsby';
-  } else if (dependencyNames.includes('react-scripts')) {
-    return 'create-react-app';
-  }
+  // Make sure to check for the existance of dependencies
+  // because sometimes package.json only has devDependencies
+  if (json.dependencies) {
+    const dependencyNames = Object.keys(json.dependencies);
+
+    if (dependencyNames.includes('gatsby')) {
+      return 'gatsby';
+    } else if (dependencyNames.includes('react-scripts')) {
+      return 'create-react-app';
+    }
 
-  // An ejected create-react-app won't have `react-scripts`.
-  // So it actually becomes kinda hard to figure out what kind of project it is!
-  // One strong clue is that it will have `eslint-config-react-app` as a
-  // dependency... this isn't foolproof since a user could easily uninstall
-  // that dependency, but it'll work for now.
-  // In the future, could also check the `config` dir for the standard React
-  // scripts
-  if (dependencyNames.includes('eslint-config-react-app')) {
-    return 'create-react-app';
+    // An ejected create-react-app won't have `react-scripts`.
+    // So it actually becomes kinda hard to figure out what kind of project it is!
+    // One strong clue is that it will have `eslint-config-react-app` as a
+    // dependency... this isn't foolproof since a user could easily uninstall
+    // that dependency, but it'll work for now.
+    // In the future, could also check the `config` dir for the standard React
+    // scripts
+    if (dependencyNames.includes('eslint-config-react-app')) {
+      return 'create-react-app';
+    }
```

Comment:
It could just be me, but I really like when "disqualifying" concerns, like not having a `dependencies` key, are dealt with through early returns:

```
const inferProjectType = json => {
  if (!json.dependencies) {
    return null;
  }

  const dependencyNames = Object.keys(json.dependencies);
  ...
}
```

I like this because I feel like it frees your mind from worrying about that condition when you're reading the rest of the logic. It's not nested within a conditional that you have to keep in your head; instead it's just the core logic for qualified inputs.

That said, I recognize this as the subjective type of thing people care about more than they should, haha. So it's not worth blocking the review over! I'll quickly throw in a commit that does this change after the merge. Just explaining so you don't wonder why your code changed so quick after merging :)

Reply:
I think that definitely reads easier 👍